crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -667,18 +670,31 @@ impl<'db> TypeInferenceBuilder<'db> {
     }
 
     fn infer_name_expression(&mut self, name: &ast::ExprName) -> Type<'db> {
-        let ast::ExprName {
-            range: _,
-            id: _,
-            ctx,
-        } = name;
+        let ast::ExprName { range: _, id, ctx } = name;
 
         match ctx {
             ExprContext::Load => {
-                let use_def = self.index.use_def_map(self.scope.file_scope_id(self.db));
+                let file_scope_id = self.scope.file_scope_id(self.db);
+                let use_def = self.index.use_def_map(file_scope_id);
                 let use_id = name.scoped_use_id(self.db, self.scope);
                 let definitions = use_def.use_definitions(use_id);
-                definitions_ty(self.db, definitions, use_def.use_may_be_unbound(use_id))
+                let may_be_unbound = use_def.use_may_be_unbound(use_id);
+                let symbols = self.index.symbol_table(file_scope_id);
+                // SAFETY: the symbol table always creates a symbol for every Name node.
+                let symbol = symbols.symbol_by_name(id).unwrap();
```

Comment:
```suggestion
                let symbol = symbols.symbol_by_name(id)
                    .expect("Expected the symbol table to always crate a symbol for every Name node");
```

You should also remove the `allow(unused)` suppression in `symbols.rs`:

https://github.com/astral-sh/ruff/blob/648cca199bbf0d2b4b04414404bca866fe5b52c9/crates/red_knot_python_semantic/src/semantic_index/symbol.rs#L195-L200